src/v3/plugins/github/urlIdParse.test.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import {
+  commentUrlToId,
+  reviewUrlToId,
+  reviewCommentUrlToId,
+} from "./urlIdParse";
+
+describe("plugins/github/urlToId", () => {
+  const issueComment =
+    "https://github.com/sourcecred/sourcecred/issues/350#issuecomment-394939349";
+  const pullComment =
+    "https://github.com/sourcecred/sourcecred/pull/363#issuecomment-395836900";
+  const reviewComment =
+    "https://github.com/sourcecred/sourcecred/pull/380#discussion_r194816899";
+  const review =
+    "https://github.com/sourcecred/sourcecred/pull/383#pullrequestreview-128199239";
+
+  describe("works correctly", () => {
+    it("for issueComment", () => {
+      expect(commentUrlToId(issueComment)).toEqual("394939349");
+    });
+    it("for pullComment", () => {
+      expect(commentUrlToId(pullComment)).toEqual("395836900");
+    });
+    it("for reviewComment", () => {
+      expect(reviewCommentUrlToId(reviewComment)).toEqual("194816899");
+    });
+    it("for review", () => {
+      expect(reviewUrlToId(review)).toEqual("128199239");
+    });
+  });
+  describe("errors on", () => {
+    const bad = [
+      {f: commentUrlToId, instances: [reviewComment, review]},
+      {f: reviewCommentUrlToId, instances: [pullComment, issueComment, review]},
+      {f: reviewUrlToId, instances: [pullComment, issueComment, reviewComment]},
+    ];
+    bad.forEach(({f, instances}) => {
+      it(`${f.name} receiving the wrong url type`, () => {
+        instances.forEach((i) => {
```

Comment:
It would be nice if this could indicate which example failed (e.g., by
using separate `it` blocks). If that’s not too hard to implement, I’d
prefer it.

Reply:
Done. I also added checks on null, undefined, and a garbage string.